android/src/main/java/io/event1/capacitorstripeterminal/StripeTerminal.java

```diff
@@ -936,6 +938,38 @@ public void onFailure(@NonNull TerminalException e) {
     }
   }
 
+  @PluginMethod
+  public void tapToPaySupported(final PluginCall call) {
```

Comment:
Yea, unfortunately this is the official Stripe way of checking.

In this instance I have added a value that is set on first run of the method call, in order to speed up subsequent calls without requiring a call to `discoverReaders`. Although not optimal, scanning for the compatibility/LocalMobileReader is fast (generally less than a second). 

I have also added a call to cancel `discoverReaders` if it's already running and I have also added a call to the `pendingDiscoverReaders.cancel()` method within `cancelDiscoverReaders` before nulling `pendingDiscoverReaders` as Stripe includes calling that method in their docs.

If there's anything further that you would like me to change, let me know. I happy to remove this method entirely if it irks you, as it is more a nice to have check rather than a requirement.